src/services/stac/index.tsx

```diff
@@ -104,12 +103,14 @@ const getStacCatalogUrl = (collection: Collection): string => {
   const selfLink = collection.links.find((link) => link.rel === 'self');
   if (!selfLink?.href) return '';
 
-  let url = selfLink.href.replace(import.meta.env.VITE_STAC_ENDPOINT, '');
-  if (url.startsWith('/catalogs/')) {
-    url = url.replace('/catalogs/', '');
-  }
+  // Extract central part of URL, so take:
+  // `https://test.eodatahub.org.uk/api/catalogue/stac/catalogs/supported-datasets/ceda-stac-fastapi/collections/cmip6`
+  // and return `supported-datasets/ceda-stac-fastapi`.
+  const catalogsStr = '/catalogs/';
+  const start = selfLink.href.indexOf(catalogsStr) + catalogsStr.length;
+  const end = selfLink.href.indexOf('/collections/');
+  const catalogUrl = selfLink.href.slice(start, end);
```

Comment:
So this takes the first instance of `/catalogs/` and returns the part of the path after. There could be instances of collections that have catalogs within a collection of another catalog. 

Lets take for example this URL:
`https://test.eodatahub.org.uk/api/catalogue/stac/catalogs/supported-datasets/ceda-stac-fastapi/catalogs/test-catalog/collections/test-collection/catalogs/test-catalog-2/collections/test-collection-2`

So breaking this down. We have the `supported-datasets` catalog that contains a catalog called `ceda-stac-fastapi`.
The `ceda-stac-fastapi` has a nested catalog called `test-catalog` that contains a collection called `test-collection`.
This also has even more catalogs, one of them being `test-catalog-2` that ultimately has a collection called `test-collection-2`. 

With that explained lets now then turn your code into a function lets say:
```javascript
const extractUrl = (selfLink) => {
  const catalogsStr = '/catalogs/';
  const start = selfLink.href.indexOf(catalogsStr) + catalogsStr.length;
  const end = selfLink.href.indexOf('/collections/');
  const catalogUrl = selfLink.href.slice(start, end);
  return catalogUrl;
}
```
And then call `extractUrl` with this catalog:
```javascript
const link = {
    href: 'https://test.eodatahub.org.uk/api/catalogue/stac/catalogs/supported-datasets/ceda-stac-fastapi/catalogs/test-catalog/collections/test-collection/catalogs/test-catalog-2/collections/test-collection-2'
}
```

When we call that function using `extractUrl(link)` we will get:
`'supported-datasets/ceda-stac-fastapi/catalogs/test-catalog'`

Which is not the correct catalog to fetch the items from.